server/src/main/java/org/opensearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -364,13 +366,14 @@ private void checkSystemIndexAccess(Context context, Metadata metadata, Set<Inde
                 .map(i -> i.getIndex().getName())
                 .sorted() // reliable order for testing
                 .collect(Collectors.toList());
-            if (resolvedSystemIndices.isEmpty() == false) {
+            if (resolvedSystemIndices.isEmpty() == false && !isDeprecationWarningAlreadyLogged) {
```

Comment:
This has a potential race condition where two threads may end up logging and then flipping the value, no?